### PR TITLE
Allow leader election controls for manager

### LINKED ISSUE #4147: Seldon core operator is restarting due to failed renewal of lease
## Describe the bug
Seldon core operator pod is getting restarted due to failed to retrieve resource lock getting below error logs.

`{"version": "1.0", "level": "INFO", "host": "ccs-seldon-75dfcb5bf9-bwkfv.ccs", "system": "ml-inf-seldon", "type": "log", "log": {"message": "E0610 16:36:08.554283       7 leaderelection.go:325] error retrieving resource lock ccs/a33bd623.machinelearning.seldon.io: Get \"https://10.254.0.1:443/apis/coordination.k8s.io/v1/namespaces/ccs/leases/a33bd623.machinelearning.seldon.io\": context deadline exceeded"}, "time": "2022-06-10T16:36:09.511Z"}
{"version": "1.0", "level": "INFO", "host": "ccs-seldon-75dfcb5bf9-bwkfv.ccs", "system": "ml-inf-seldon", "type": "log", "log": {"message": "I0610 16:36:08.554523       7 leaderelection.go:278] failed to renew lease ccs/a33bd623.machinelearning.seldon.io: timed out waiting for the condition"}, "time": "2022-06-10T16:36:09.512Z"}
{"version": "1.0", "level": "INFO", "host": "ccs-seldon-75dfcb5bf9-bwkfv.ccs", "system": "ml-inf-seldon", "type": "log", "log": {"message": "setup : problem running manager"}, "time": "2022-06-10T16:36:09.512Z"}
`
Wanted to get more insights on this issue and is this issue is related ( https://github.com/kubernetes/client-go/issues/966 )

## To reproduce
1. Install seldon chart with 2 replicas, keep it active for 2-3 days one or more restarts we will see.

## Expected behaviour
Seldon pod must not restart it should retry for lease renewal.

## Environment

* Cloud Provider: Bare Metal
* Kubernetes Cluster Version: `[root:ccs-01-control-01 /root]$ kubectl version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.5", GitCommit:"e338cf2c6d297aa603b50ad3a301f761b4173aa6", GitTreeState:"clean", BuildDate:"2020-12-09T11:18:51Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.5", GitCommit:"e338cf2c6d297aa603b50ad3a301f761b4173aa6", GitTreeState:"clean", BuildDate:"2020-12-09T11:10:32Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}`

* Deployed Seldon System Images: v1.11.0

## Model Details 
* Images of your model: NA
* Logs of your model: NA

### PR DESCRIPTION
**Pull request by cliveseldon**
**What this PR does / why we need it**:
 * Allows leadership election controls for manager. This will allow more control for devops in clusters that may have noisy/slow leadership election scenarios.

**Which issue(s) this PR fixes**:

Fixes #4147 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **cliveseldon**
You can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: helm-charts/seldon-core-operator/templates/webhook.yaml [hunk 1]
```diff
@@ -34,9 +20,7 @@ webhooks:
   - v1
   - v1beta1
   clientConfig:
-{{- if not .Values.certManager.enabled }}
```
Comment (axsaucedo) [inline helm-charts/seldon-core-operator/templates/webhook.yaml:37]: It seems this was accidentally removed, may be best to just revert the webhook.yaml diff

Comment (cliveseldon) [reply helm-charts/seldon-core-operator/templates/webhook.yaml:37]: This is automatically created by the python script so would be a bug if it was removed?

Comment (axsaucedo) [reply helm-charts/seldon-core-operator/templates/webhook.yaml:37]: Yeah it definitely would be a bug in the python helm generators - we could fix it as a separate pr or if it's a quick fix could just be updated here

Comment (cliveseldon) [reply helm-charts/seldon-core-operator/templates/webhook.yaml]: I reverted for now and added an issue #4217

Comment (axsaucedo) [reply helm-charts/seldon-core-operator/templates/webhook.yaml]: https://github.com/SeldonIO/seldon-core/pull/4218/files

**Comment by axsaucedo** [comment]
Only test fail is confirmed flaky - moving ahead with merge
/approve

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/4211#issuecomment-1184470336 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**